### PR TITLE
add image pull secrets to product-jenkins

### PR DESCRIPTION
**Pull request by mrparkers**
turns out we can just use the docker config json we were already building for this

### CODE REVIEW

File: modules/lead/product-jenkins/dockercfg.tf [hunk 1]
```diff
@@ -34,3 +34,31 @@ resource "kubernetes_secret" "jenkins_repository_dockercfg" {
...
+
+  data = {
+    ".dockerconfigjson" = data.template_file.dockercfg.rendered
+  }
+}
+
+resource "kubernetes_secret" "staging_image_pull_secrets" {
+  provider = kubernetes.staging
+  metadata {
+    name      = "image-pull-secret"
```
Comment (gesparza3) [inline modules/lead/product-jenkins/dockercfg.tf:55]: So to get his working we'll just have to have our deployment point to this new secret?

Comment (mrparkers) [reply modules/lead/product-jenkins/dockercfg.tf:55]: that's the idea

**Review by gesparza3 — APPROVED** [review]
resources looks good to me, approving now. Should probably wait to merge until functionality is tested

**Review by RobertKelly — APPROVED** [review]

File: modules/lead/product-jenkins/pipelines-github.tpl [hunk 2]
```diff
@@ -0,0 +1,22 @@
+{"jobs":[%{ for name,pipeline in pipelines ~}{"script":"
+    folder('${pipeline.org}') {
+    }
+    multibranchPipelineJob('${pipeline.org}/${pipeline.repo}') {
+        triggers {
+            periodic(1)
+        }
+        branchSources {
+            github {
```
Comment (RobertKelly) [inline modules/lead/product-jenkins/pipelines-github.tpl:9]: Looks good

File: modules/lead/product-jenkins/variables.tf [hunk 3]
```diff
@@ -34,3 +34,6 @@ variable "pipelines" {
 
 variable "vault_namespace" {}
 variable "vault_root_token_secret" {}
+variable "jenkins_pipeline_source" {
+  default = "git"
```
Comment (RobertKelly) [inline modules/lead/product-jenkins/variables.tf:38]: Saw `github` used earlier. Is `git` what we want to default to?

Comment (mrparkers) [reply modules/lead/product-jenkins/variables.tf:38]: yeah, that's what we will continue to use for ignite labs

**Pull request merged**